### PR TITLE
fix: Add missing 'archived' prop for ArtifactPanel component. Fixes #12331

### LINKED ISSUE #12331: Artifact in workflow diagram not working when workflow is archived and deleted from cluster
### Pre-requisites

- [X] I have double-checked my configuration
- [X] I can confirm the issues exists when I tested with `:latest`
- [ ] I'd like to contribute the fix myself (see [contributing guide](https://github.com/argoproj/argo-workflows/blob/main/docs/CONTRIBUTING.md))

### What happened/what did you expect to happen?

Artifact URLs used in the workflow diagram's artifact icons seem to no longer work once a workflow is archived and deleted from the cluster (Returns Internal Server Error). However, if you click on the node from which the artifact is output from and attempt to download the artifact from the right-hand bar's input/output's menu then the artifact successfully downloads.

Difference in the 2 links:

link from workflow diagram's artifact icon: `https://<argo-host>/artifact-files/<namespace>/workflows/<workflow-name>/<workflow-name>-1855980756/outputs/result_html`

link from node's input/output menu: `https://<argo-host>/artifact-files/<namespace>/archived-workflows/f7655950-a426-4fb3-a82d-eb1f53dfb948/<workflow-name>-1855980756/outputs/result_html`

As you can see the link in the workflow diagram's artifact icon is still trying to reference the worklow as a non-archived workflow. Whereas the link in the node's input/output menu is now referencing it as an archived-workflow in the API. I suspect this is causing the argo-server to return Internal Server Error when clicking on the link from the workflow diagram's artifact icon.

**NOTE**: In order to reproduce the workflow and its artifacts must be archived AND deleted from the cluster in order to reproduce

### Version

v3.5.2

### Paste a small workflow that reproduces the issue. We must be able to run the workflow; don't enter a workflows that uses private images.

```YAML
apiVersion: argoproj.io/v1alpha1
kind: Workflow
metadata:
  generateName: artifact-passing-
spec:
  entrypoint: artifact-example
  templates:
  - name: artifact-example
    steps:
    - - name: generate-artifact
        template: whalesay
    - - name: consume-artifact
        template: print-message
        arguments:
          artifacts:
          # bind message to the hello-art artifact
          # generated by the generate-artifact step
          - name: message
            from: "{{steps.generate-artifact.outputs.artifacts.hello-art}}"

  - name: whalesay
    container:
      image: docker/whalesay:latest
      command: [sh, -c]
      args: ["cowsay hello world | tee /tmp/hello_world.txt"]
    outputs:
      artifacts:
      # generate hello-art artifact from /tmp/hello_world.txt
      # artifacts can be directories as well as files
      - name: hello-art
        path: /tmp/hello_world.txt

  - name: print-message
    inputs:
      artifacts:
      # unpack the message input artifact
      # and put it at /tmp/message
      - name: message
        path: /tmp/message
    container:
      image: alpine:latest
      command: [sh, -c]
      args: ["cat /tmp/message"]
```

### Logs from the workflow controller

```text
N/A
```

### Logs from in your workflow's wait container

```text
N/A
```

### PR DESCRIPTION
**Pull request by Garett-MacGowan**
Fixes #12331

### Motivation

A missing `archived` prop in the UI's `ArtifactPanel` component was causing an improper URL to be used to fetch the artifact (assuming the workflow was archived).

### Modifications

Added the missing `archived` prop to the `ArtifactPanel` component.

### Verification

I added tests to the argo server test suite to account for the URL structure which was previously untested. I called this test `GetArtifactByNodeID`, in accordance with naming conventions in similar tests.

For the UI, I ran the `TestArtifactServerArtifactPassing` test (which contains the `GetArtifactByNodeID` test mentioned above) using:
```
make start UI=true PROFILE=postgres
make TestArtifactServerArtifactPassing
```
Once it completed, I manually verified that the artifact in the ArtifactPanel was downloadable.

image

image

(download button)
image

### CODE REVIEW

**Comment by jmeridth** [comment]
Screenshot example?

**Comment by Garett-MacGowan** [comment]
> Screenshot example?

I'll provide it in a moment.

**Comment by Garett-MacGowan** [comment]
> Screenshot example?

@jmeridth I added the screenshots.

**Comment by Garett-MacGowan** [comment]
@agilgur5 this is rebased and ready to go

File: test/e2e/argo_server_test.go [hunk 1]
```diff
@@ -1037,6 +1037,30 @@ spec:
 	})
 }
 
+func (s *ArgoServerSuite) TestArtifactServerArtifactPassing() {
```
Comment (agilgur5) [inline test/e2e/argo_server_test.go:1040]: I feel like we should specify that this checking Archived Workflows specifically, though I do see that some of the tests name it explicitly and others don't 😕

**Review by agilgur5 — APPROVED** [review]
Just have a small comment around test naming, but otherwise LGTM.

Great catch and thanks for the detailed fix!

**Comment by agilgur5** [comment]
Also is there a reason you merge `main` frequently in this PR? We don't require that PRs be up-to-date with `main` so it's not necessary. For most PRs, it usually doesn't affect anything either.

**Comment by agilgur5** [comment]
Btw @Garett-MacGowan would you be interested in [joining the Argo project as a Member](https://github.com/argoproj/argo-workflows/blob/a6b1217ac47fa30e27db4f0a0669b477586fd60d/docs/CONTRIBUTING.md#roles)?
You've had some great fixes along with very high quality PRs etc and we'd love to have you if you're interested. I would be willing to sponsor and I think Julie and others would as well 🙂

File: test/e2e/argo_server_test.go (hunk 1 above)
Comment (Garett-MacGowan) [reply test/e2e/argo_server_test.go:1040]: Will do.

**Comment by Garett-MacGowan** [comment]
> Btw @Garett-MacGowan would you be interested in [joining the Argo project as a Member](https://github.com/argoproj/argo-workflows/blob/a6b1217ac47fa30e27db4f0a0669b477586fd60d/docs/CONTRIBUTING.md#roles)? You've had some great fixes along with very high quality PRs etc and we'd love to have you if you're interested. I would be willing to sponsor and I think Julie and others would as well 🙂

I'd be happy to! There's no strict hourly commitment right? I'd be happy to spend a few hours every week.

**Comment by Garett-MacGowan** [comment]
> Also is there a reason you merge `main` frequently in this PR? We don't require that PRs be up-to-date with `main` so it's not necessary. For most PRs, it usually doesn't affect anything either.

Didn't know PRs didn't need to be up to date. If there are conflicts, it would provide a different prompt than "Update branch" right?

File: test/e2e/argo_server_test.go (hunk 1 above)
Comment (Garett-MacGowan) [reply test/e2e/argo_server_test.go]: @agilgur5 I changed the name to `TestArtifactServerArchivedWorkflow`. Is this more along the lines of what you were thinking?

**Comment by agilgur5** [comment]
> Didn't know PRs didn't need to be up to date.

It is a requirement you can add on GitHub, but we don't currently require it. There's only been a handful of times where that's caused an issue, which is actually with "invisible" merge conflicts (two PRs that when merged together break a build, but do not conflict in version control itself). I think we can also solve those with a merge queue as well instead of an extra rule, as was discussed briefly [over Slack](https://cloud-native.slack.com/archives/C0510EUH90V/p1701288620675009?thread_ts=1700702475.578759&cid=C0510EUH90V)

> If there are conflicts, it would provide a different prompt than "Update branch" right?

Yep, it would say merging is not possible due to conflicts. The prompt instead says "Resolve conflicts" in that case.

**Comment by agilgur5** [comment]
> @agilgur5 I'd be happy to! There's no strict hourly commitment right? I'd be happy to spend a few hours every week.

Awesome! Nope, there are currently no strict commitments associated with [Membership or other roles](https://github.com/argoproj/argoproj/blob/main/community/membership.md). It comes with some additional privileges (i.e. issue triage) and some small responsibilities (that are usually the norm for active contributors in most repos anyway). Could you file a membership request issue in the `argoproj` repo as is described in the above doc? Julie upvoted my comment so can list both of us as sponsors and I'm pretty sure the Leads would be glad to sponsor as well once they see the request 🙂

We do also have a _separate_ [Sustainability Effort](https://github.com/argoproj/argo-workflows/blob/main/community/sustainability_effort.md) that was recently created that is specific to Argo Workflows (i.e. does not apply to CD or other projects at this time) that is a completely optional addition to the formal Argo project roles. If you're interested in that as well, that _does_ have an hourly commitment but is not "strict" per se, as it is an average over time.

File: test/e2e/argo_server_test.go (hunk 1 above)
Comment (agilgur5) [reply test/e2e/argo_server_test.go]: Works for me. Naming is [hard](https://martinfowler.com/bliki/TwoHardThings.html), especially for this level of specificity 😅

**Comment by Garett-MacGowan** [comment]
@agilgur5, @juliev0 can this be merged? It's unfortunate it didn't make it into the latest release.

**Review by juliev0 — APPROVED** [review]

**Comment by juliev0** [comment]
> For reference, I'm [not an Approver](https://github.com/argoproj/argo-workflows/blob/44b33fadf2d2d73cde2b5965c4a05946e45b6e75/OWNERS#L13) (yet), so I don't have merge permissions, which is why I requested a review from Julie.
> 
> Agreed this would have been good to have for the release, but we'll have another soon enough since there are still two P1 bugs out.

Sorry, I missed that you requested my review!

**Comment by Garett-MacGowan** [comment]
> > For reference, I'm [not an Approver](https://github.com/argoproj/argo-workflows/blob/44b33fadf2d2d73cde2b5965c4a05946e45b6e75/OWNERS#L13) (yet), so I don't have merge permissions, which is why I requested a review from Julie.
> > Agreed this would have been good to have for the release, but we'll have another soon enough since there are still two P1 bugs out.
> 
> Sorry, I missed that you requested my review!

Thanks Julie!

**Pull request merged**